### PR TITLE
[libical] Add new port

### PR DESCRIPTION
**Pull request by winterz**
This is a new port for the libical project and also my first ever attempt to create a vcpkg port.

### CODE REVIEW

File: ports/libical/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+
+file(COPY ${CURRENT_PACKAGES_DIR}/lib/cmake/LibICal
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libical/cmake
+    FILES_MATCHING PATTERN "*.cmake" )
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/cmake)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/cmake)
```
Comment (NancyLi1013) [inline ports/libical/portfile.cmake]: Please remove these lines
```
file(COPY ${CURRENT_PACKAGES_DIR}/lib/cmake/LibICal
    DESTINATION ${CURRENT_PACKAGES_DIR}/share/libical/cmake
    FILES_MATCHING PATTERN "*.cmake" )
file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/cmake)
file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/cmake)
```
and use this `vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/LibIcal TARGET_PATH share/LibIcal)`.

File: ports/libical/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/libical)
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libical/LICENSE.txt)
+file(COPY ${SOURCE_PATH}/LICENSE.LGPL21.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/libical)
+file(COPY ${SOURCE_PATH}/LICENSE.MPL2.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/libical)
+file(RENAME ${CURRENT_PACKAGES_DIR}/share/libical/COPYING ${CURRENT_PACKAGES_DIR}/share/libical/copyright)
```
Comment (NancyLi1013) [inline ports/libical/portfile.cmake]: Please keep one of licenses and remove others. 
f`ile(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT }RENAME copyright)`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @winterz  thanks for this pull request. Please try to sign CLA first.  You can click the following link to complete it.
https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8754

File: ports/libical/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,46 @@
+include(vcpkg_common_functions)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO libical/libical
+    REF v3.0.6
```
[the remark this answers is not in the document]
Comment (winterz) [reply ports/libical/portfile.cmake]: ok, but tags are convenient and easy to read and understand.
Can I put in a comment saying "this is tag v3.0.6" ?

File: ports/libical/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,46 @@
+include(vcpkg_common_functions)
```
[the remark this answers is not in the document]
Comment (winterz) [reply ports/libical/portfile.cmake]: ok, but the examples need to be modified as well.

File: ports/libical/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO libical/libical
+    REF v3.0.6
+    SHA512 624b70b54da1ff30085fd16bf136496f179088dd089a3bea7e9cfcf3ad07588907d99ff07782b33f5fdfd4fbfc0bae9daa42bb965380b28bf1c9ce4481b698cb
+)
+
+#Perl is required by the buildsystem.
+vcpkg_find_acquire_program(PERL)
+
```
[the remark this answers is not in the document]
Comment (winterz) [reply ports/libical/portfile.cmake]: will do.

File: ports/libical/portfile.cmake (hunk 1 above)
Comment (winterz) [reply ports/libical/portfile.cmake]: nice!

**Comment by winterz** [comment]
I will update the pull request as indicated.

as for signing the CLA: my employer is ok with my signing it, but they are deciding on a company-wide policy.  might take some time.

**Comment by winterz** [comment]
same to you bub :)
I'm not sure what you're referring to @JackBoosY

**Comment by JackBoosY** [comment]
@winterz I just restart the CI test.

**Comment by NancyLi1013** [comment]
@winterz 
I also noticed that you have added these two ports `qt5-kdreports` and `qt5-kdsoap` in this PR.
Could you add them in a separate PR? Please refer to this [doc](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md).

As `qt5-kdreports` and `qt5-kdsoap` belong to qt5. You can refer to other qt5 components to add them.
So please remove these two ports from this PR.

**Comment by winterz** [comment]
I'm going to abandon this for now.

**Pull request closed without merging**